seed/seed20.go

```diff
@@ -448,9 +640,42 @@ func (s *seed20) lookupSnap(snapRef naming.SnapRef, optSnap *internal.Snap20, ch
 		timings.Run(tm, "derive-side-info", fmt.Sprintf("hash and derive side info for snap %q", snapRef.SnapName()), func(nested timings.Measurer) {
 			var snapRev *asserts.SnapRevision
 			var snapDecl *asserts.SnapDeclaration
-			path, snapRev, snapDecl, err = s.lookupVerifiedRevision(snapRef, handler, snapsDir, tm)
-			if err == nil {
-				sideInfo = snapasserts.SideInfoFromSnapAssertions(snapDecl, snapRev)
+			path, snapRev, snapDecl, err = s.lookupVerifiedRevision(
+				snapRef, handler, snapsDir, tm)
+			if err != nil {
+				return
+			}
+			sideInfo = snapasserts.SideInfoFromSnapAssertions(snapDecl, snapRev)
+
+			if modelSnap != nil {
```

Comment:
don't we need to deal with added extra snaps in dangerous mode that are asserted? those could also have components that would be asserted but not in the model?